Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -256,6 +272,34 @@ public void onError()
 
 	}
 
+	@Override
+	protected void onPause()
+	{
+		if (!isManualExit)
+		{
+			mEmulationFragment.saveTemporaryState();
+		}
```

Comment:
Android documentation always recommend to save persistent data inside onPause and not onSaveInstanceState, but after going through the documentation again it seems the only reason for that is what you described since onSaveInstanceState will not be called on manual exist. I guess I can try that out but I will need another round of testing. Android activity life cycle always has it's own surprises :)